### ISSUE TITLE
Possible bug when setting k > 1 & Gumbel_hard = True

### ISSUE THREAD

**Issue by hadyelsahar** [sosuperic/MeanSum#13]
Thanks for the interesting work and code

I was trying to get my head around the code and I couldn't understand something:

When training the `mlstm` model If we try the following set of parameters:  
* gumbel_hard = true 
* sampling method = "greedy" or "sample"
* k > 1   

In the line [mlstm #L291](https://github.com/sosuperic/MeanSum/blob/master/models/mlstm.py#L291)
The `logits_to_prob` function will return a strict one hot vector according to the Torch gumbel softmax implementation `F.gumbel_softmax`

Afterwards, this prob vector is sent to `prob_to_vocab_id` method which is supposed to apply either `torch.top_k` (beam search)   or  `torch.multinomial` (top k sampling).

Implementation wise this shouldn't show any errors in beam search because of the `torch.topk` function ability to handle draws, however, the top k you get aren't the actual top k probabilities e.g. 

![image](https://user-images.githubusercontent.com/1453243/61800378-335cec80-ae2d-11e9-89ea-932f723174e4.png)

But if you try to sample `multinomial` from 1 hot vector where K > 1 you get a runtime error:
![image](https://user-images.githubusercontent.com/1453243/61800533-7fa82c80-ae2d-11e9-9516-43ded07462b1.png)

Am I missing something here?
